lib/core/request.js

```diff
@@ -335,7 +323,7 @@ function processHeader (request, key, val, skipAppend = false) {
     const arr = []
     for (let i = 0; i < val.length; i++) {
       if (typeof val[i] === 'string') {
-        if (headerCharRegex.exec(val[i]) !== null) {
+        if (!util.isValidHeaderChar(val[i])) {
```

Comment:
can we have named imports instead of default import?

Reply:
it'll be slower on longer headers and faster on shorter, a benchmark for every utility function isn't needed

Reply:
although it would be nice to see if this actually does improve performance, and if it does, maybe upstream it to core

Reply:
a benchmark would help to verify the proposal. e.g. using exec means instantiating an object to return. So why was the regex not using .test(), which always returns true and false?

Reply:
> So why was the regex not using .test(), which always returns true and false?

I think this is because exec is faster when primordials are used.